observer/src/observer/cli.py

```
@@ -0,0 +1,291 @@
+        click.echo("  Checking pgvector...", nl=False)
+        with engine.connect() as conn:
+            row = conn.execute(
+                text("SELECT 1 FROM pg_available_extensions WHERE name = 'vector'")
+            ).fetchone()
+            if row is None:
+                click.echo(" NOT FOUND")
+                click.echo(
+                    "  pgvector is not installed on this PostgreSQL server.\n"
+                    "  Install it first: https://github.com/pgvector/pgvector#installation"
+                )
+                raise SystemExit(1)
```

Ruff (0.15.1)

[warning] Abstract `raise` to an inner function

(TRY301)